src/shared/actions/finalize-message-signature.ts

```diff
@@ -0,0 +1,43 @@
+import { ExternalMethods, MESSAGE_SOURCE, SignatureResponseMessage } from '@shared/message-types';
+import { logger } from '@shared/logger';
+import { SignatureData } from '@stacks/connect';
+
+interface FormatMessageSigningResponseArgs {
+  request: string;
+  response: SignatureData | string;
+}
+export function formatMessageSigningResponse({
+  request,
+  response,
+}: FormatMessageSigningResponseArgs): SignatureResponseMessage {
+  return {
+    source: MESSAGE_SOURCE,
+    method: ExternalMethods.signatureResponse,
+    payload: { signatureRequest: request, signatureResponse: response },
+  };
+}
+
+interface FinalizeMessageSignatureArgs {
+  requestPayload: string;
+  tabId: number;
+  data: SignatureData | string;
+}
+export function finalizeMessageSignature({
+  requestPayload,
+  data,
+  tabId,
+}: FinalizeMessageSignatureArgs) {
+  try {
+    const responseMessage = formatMessageSigningResponse({
+      request: requestPayload,
+      response: data,
+    });
+    chrome.tabs.sendMessage(tabId, responseMessage);
+    window.close();
+  } catch (error) {
+    logger.debug('Failed to get Tab ID for message signature request:', requestPayload);
+    throw new Error(
+      'Your message was signed, but we lost communication with the app you started with.'
```

Comment:
I think there was feedback this error msg is unclear, should we change here?

Reply:
Glad you mention, and [kinda why I asked this q earlier ](https://discord.com/channels/621759717756370964/745197302255321108/1007207189414424576). I never really understood what it meant, and now there can never not be a tabId 🤔

Reply:
I've killed these message on all but tx, where we're tracking whether or not this err is thrown

Reply:
I am trying to think of a better error msg. 🤔 

'Your message was signed, but the app tab initiating this transaction was closed'

Reply:
Presumably this error is shown when the wallet can't pass the signed message back to the app since it's (likely) been closed. As such, the signed message is basically useless for the user. Correct?

If so, the copy could be:

`We're unable to send this message back to the app used to initiate its signing, perhaps because you've closed it. Please ensure the app is open and reinitiate the signing process.`

Reply:
Sending a msg to a non-existent tab doesn't throw an error, which is why this confused me; even when that happens it shouldn't have triggered this error. And we don't have any further checking, so not sure when this would've fired. Added analytics to check